### PR TITLE
fixed Stationeers for new Server

### PR DESCRIPTION
**Pull request by gOOvER**
# Description

Complete rewrite of the Stationeers Egg, because fixing the old one, would be more work, than creating a new one.

## Changelog
- switch to PTDL_v2
- Startup fixed !!! CONSOLE DOES NOT REALLY WORK IN THE NEW SERVER
- updated all images
- updated all Scripts
- add some more Variables 

Fixes: #1752 

PR that can be closed: #1763 

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?

### CODE REVIEW

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

**Comment by QuintenQVD0** [comment]
It looks fine to me but stil need to test it or someone external to leave a comment with proof that it is working

File: game_eggs/steamcmd_servers/stationeers/egg-stationeers.json [hunk 1]
```diff
@@ -1,61 +1,74 @@
...
.../steamcmd_linux.tar.gz\r\ntar -xzvf steamcmd.tar.gz -C \/mnt\/server\/steamcmd\r\nmkdir -p \/mnt\/server\/steamapps # Fix steamcmd disk write error when this folder is missing\r\ncd \/mnt\/server\/steamcmd\r\n\r\n# SteamCMD fails otherwise for some reason, even running as root.\r\n# This is changed at the end of the install process anyways.\r\nchown -R root:root \/mnt\r\nexport HOME=\/mnt\/server\r\n\r\n## install game using steamcmd\r\n.\/steamcmd.sh +force_install_dir \/mnt\/server +login ${STEAM_USER} ${STEAM_PASS} ${STEAM_AUTH} $( [[ \"${WINDOWS_INSTALL}\" == \"1\" ]] && printf %s '+@sSteamCmdForcePlatformType windows' ) +app_update ${SRCDS_APPID} $( [[ -z ${SRCDS_BETAID} ]] || printf %s \"-beta ${SRCDS_BETAID}\" ) $( [[ -z ${SRCDS_BETAPASS} ]] || printf %s \"-betapassword ${SRCDS_BETAPASS}\" ) ${INSTALL_FLAGS} validate +quit ## other flags may be needed depending on install. looking at you cs 1.6\r\n\r\n## set up 32 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk32\r\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so\r\n\r\n## set up 64 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so",
```
[the remark this answers is not in the document]
Comment (gOOvER) [reply game_eggs/steamcmd_servers/stationeers/egg-stationeers.json]: done

**Comment by kkoshelev** [comment]
> It looks fine to me but stil need to test it or someone external to leave a comment with proof that it is working

I tried it today. Server doesn't install properly or so it seems.

```
Using anonymous user.
steamcmd.sh
linux32/steamcmd
linux32/steamerrorreporter
linux32/libstdc++.so.6
linux32/crashhandler.so
Redirecting stderr to '/mnt/server/Steam/logs/stderr.txt'
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[  0%] Checking for available update...
KeyValues Error: LoadFromBuffer: missing {   (current key: '<!DOCTYPE') in file manifest [offset: 20]
Bad expression token: tBad expression token: rBad expression token: [Bad expression token: [Bad expression token: [Bad expression token: ;Bad expression token: ;Bad expression token: bBad expression token: hBad expression token: .Bad expression token: [Bad expression token: .Bad expression token: tBad expression token: |Bad expression token: | yBad expression token: lBad expression token: tBad expression token: ;keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (25025 chars)
assert_20221202133929_5.dmp[26]: Uploading dump (out-of-process)
/tmp/dumps/assert_20221202133929_5.dmp

keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (25025 chars)
keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (25025 chars)
assert_20221202133929_5.dmp[26]: Finished uploading minidump (out-of-process): success = no

assert_20221202133929_5.dmp[26]: error: libcurl.so: cannot open shared object file: No such file or directory

assert_20221202133929_5.dmp[26]: file ''/tmp/dumps/assert_20221202133929_5.dmp'', upload no: ''libcurl.so: cannot open shared object file: No such file or directory''

Bad expression token: tBad expression token: tBad expression token: tBad expression token: tkeyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (1762 chars)
keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (1763 chars)
keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (1763 chars)
keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (1523 chars)
Bad expression token: [keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (2470 chars)

[..]

assert_20221202134000_10.dmp[33]: file ''/tmp/dumps/assert_20221202134000_10.dmp'', upload no: ''libcurl.so: cannot open shared object file: No such file or directory''

cp: cannot stat 'linux32/steamclient.so': No such file or directory
cp: cannot stat 'linux64/steamclient.so': No such file or directory
-----------------------------------------
Installation completed...
-----------------------------------------
```

after that it doesn't start

```
[..]

:/home/container$ stty cols 80 rows 24; ./rocketstation_DedicatedServer.x86_64 -loadlatest "${SAVE_NAME}" ${WORLD_NAME} -settings ServerName "${SERVER_NAME}" StartLocalHost true ServerVisible true GamePort ${SERVER_PORT} UpdatePort ${UPDATE_PORT} AutoSave ${AUTOSAVE} SaveInterval ${SAVE_INTERVAL} ServerPassword ${SERVER_PASSWD} ServerMaxPlayers ${MAX_PLAYERS} UPNPEnabled ${UPNP} ;2>&1
/entrypoint.sh: line 73: ./rocketstation_DedicatedServer.x86_64: No such file or directory
container@pterodactyl~ Server marked as offline...
```

**Comment by gOOvER** [comment]
> > It looks fine to me but stil need to test it or someone external to leave a comment with proof that it is working
> 
> I tried it today. Server doesn't install properly or so it seems.
> 
> ```
> Using anonymous user.
> steamcmd.sh
> linux32/steamcmd
> linux32/steamerrorreporter
> linux32/libstdc++.so.6
> linux32/crashhandler.so
> Redirecting stderr to '/mnt/server/Steam/logs/stderr.txt'
> ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
> [  0%] Checking for available update...
> KeyValues Error: LoadFromBuffer: missing {   (current key: '<!DOCTYPE') in file manifest [offset: 20]
> Bad expression token: tBad expression token: rBad expression token: [Bad expression token: [Bad expression token: [Bad expression token: ;Bad expression token: ;Bad expression token: bBad expression token: hBad expression token: .Bad expression token: [Bad expression token: .Bad expression token: tBad expression token: |Bad expression token: | yBad expression token: lBad expression token: tBad expression token: ;keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (25025 chars)
> assert_20221202133929_5.dmp[26]: Uploading dump (out-of-process)
> /tmp/dumps/assert_20221202133929_5.dmp
> 
> keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (25025 chars)
> keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (25025 chars)
> assert_20221202133929_5.dmp[26]: Finished uploading minidump (out-of-process): success = no
> 
> assert_20221202133929_5.dmp[26]: error: libcurl.so: cannot open shared object file: No such file or directory
> 
> assert_20221202133929_5.dmp[26]: file ''/tmp/dumps/assert_20221202133929_5.dmp'', upload no: ''libcurl.so: cannot open shared object file: No such file or directory''
> 
> Bad expression token: tBad expression token: tBad expression token: tBad expression token: tkeyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (1762 chars)
> keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (1763 chars)
> keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (1763 chars)
> keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (1523 chars)
> Bad expression token: [keyvaluessystem.cpp (321) : Assertion Failed: CKeyValuesSystem::AddStringToPool: key name too long (2470 chars)
> 
> [..]
> 
> assert_20221202134000_10.dmp[33]: file ''/tmp/dumps/assert_20221202134000_10.dmp'', upload no: ''libcurl.so: cannot open shared object file: No such file or directory''
> 
> cp: cannot stat 'linux32/steamclient.so': No such file or directory
> cp: cannot stat 'linux64/steamclient.so': No such file or directory
> -----------------------------------------
> Installation completed...
> -----------------------------------------
> ```
> 
> after that it doesn't start
> 
> ```
> [..]
> 
> :/home/container$ stty cols 80 rows 24; ./rocketstation_DedicatedServer.x86_64 -loadlatest "${SAVE_NAME}" ${WORLD_NAME} -settings ServerName "${SERVER_NAME}" StartLocalHost true ServerVisible true GamePort ${SERVER_PORT} UpdatePort ${UPDATE_PORT} AutoSave ${AUTOSAVE} SaveInterval ${SAVE_INTERVAL} ServerPassword ${SERVER_PASSWD} ServerMaxPlayers ${MAX_PLAYERS} UPNPEnabled ${UPNP} ;2>&1
> /entrypoint.sh: line 73: ./rocketstation_DedicatedServer.x86_64: No such file or directory
> container@pterodactyl~ Server marked as offline...
> ```

thats a Problem in your Server. your Install failed

File: game_eggs/steamcmd_servers/stationeers/egg-stationeers.json [hunk 2]
```diff
@@ -1,61 +1,74 @@
...
...missing\r\ncd \/mnt\/server\/steamcmd\r\n\r\n# SteamCMD fails otherwise for some reason, even running as root.\r\n# This is changed at the end of the install process anyways.\r\nchown -R root:root \/mnt\r\nexport HOME=\/mnt\/server\r\n\r\n## install game using steamcmd\r\n.\/steamcmd.sh +force_install_dir \/mnt\/server +login ${STEAM_USER} ${STEAM_PASS} ${STEAM_AUTH} $( [[ \"${WINDOWS_INSTALL}\" == \"1\" ]] && printf %s '+@sSteamCmdForcePlatformType windows' ) +app_update ${SRCDS_APPID} $( [[ -z ${SRCDS_BETAID} ]] || printf %s \"-beta ${SRCDS_BETAID}\" ) $( [[ -z ${SRCDS_BETAPASS} ]] || printf %s \"-betapassword ${SRCDS_BETAPASS}\" ) ${INSTALL_FLAGS} validate +quit ## other flags may be needed depending on install. looking at you cs 1.6\r\n\r\n## set up 32 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk32\r\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so\r\n\r\n## set up 64 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so\r\n\r\n## install end\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
```
[the remark this answers is not in the document]
Comment (gOOvER) [reply game_eggs/steamcmd_servers/stationeers/egg-stationeers.json:27]: > Is there a reason you are replacing bootstrap URL with this below? It points to an outdated binary which no longer works.
> https://steamcdn-a.akamaihd.net/client/installer/steamcmd_linux.tar.gz
> 

there is nothing outdated. thats the normal SteamCMD Script from Parker

**Comment by gOOvER** [comment]
@kkoshelev 

```
container@pterodactyl~ Starting installation process, this could take a few minutes...
steam user is not set.
Using anonymous user.
steamcmd.sh
linux32/steamcmd
linux32/steamerrorreporter
linux32/libstdc++.so.6
linux32/crashhandler.so
Redirecting stderr to '/mnt/server/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
[  0%] Downloading update...
[  0%] Checking for available updates...
[----] Download complete.
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Cleaning up...
[----] Update complete, launching Steamcmd...
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
Redirecting stderr to '/mnt/server/Steam/logs/stderr.txt'
/tmp/dumps insufficient permissions - delete and recreate
[  0%] Checking for available updates...
[----] Verifying installation...
[  0%] Downloading update...
[  0%] Checking for available updates...
[----] Download complete.
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Cleaning up...
[----] Update complete, launching Steamcmd...
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
Redirecting stderr to '/mnt/server/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation - version 1669935972
-- type 'quit' to exit --
Loading Steam API...OK
Connecting anonymously to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK
 Update state (0x5) verifying install, progress: 0.23 (6306350 / 2686736166)
 Update state (0x5) verifying install, progress: 31.44 (844637743 / 2686736166)
 Update state (0x5) verifying install, progress: 76.16 (2046117888 / 2686736166)
Success! App '600760' fully installed.
'linux32/steamclient.so' -> '../.steam/sdk32/steamclient.so'
'linux64/steamclient.so' -> '../.steam/sdk64/steamclient.so'
-----------------------------------------
Installation completed...
-----------------------------------------

```
and the start:

```
[Pterodactyl Daemon]: Checking server disk space usage, this could take a few seconds...
[Pterodactyl Daemon]: Updating process configuration files...
[Pterodactyl Daemon]: Ensuring file permissions are set correctly, this could take a few seconds...
container@pterodactyl~ Server marked as starting...
[Pterodactyl Daemon]: Pulling Docker container image, this could take a few minutes to complete...
Pulling from parkervcp/games 
Status: Image is up to date for ghcr.io/parkervcp/games:source 
Digest: sha256:2f5bb875c89ba581d2b63d7c7bdfb75b4ee21dcc28fe30a5bb3d94821f34ffdc 
[Pterodactyl Daemon]: Finished pulling Docker container image
steam user is not set.
Using anonymous user.
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
Redirecting stderr to '/home/container/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation - version 1669935972
-- type 'quit' to exit --
Loading Steam API...OK
Connecting anonymously to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK
Success! App '600760' already up to date.
:/home/container$ stty cols 80 rows 24; ./rocketstation_DedicatedServer.x86_64 -loadlatest "${SAVE_NAME}" ${WORLD_NAME} -settings ServerName "${SERVER_NAME}" StartLocalHost true ServerVisible true GamePort ${SERVER_PORT} UpdatePort ${UPDATE_PORT} AutoSave ${AUTOSAVE} SaveInterval ${SAVE_INTERVAL} ServerPassword ${SERVER_PASSWD} ServerMaxPlayers ${MAX_PLAYERS} UPNPEnabled ${UPNP} ;2>&1
[UnityMemory] Configuration Parameters - Can be set up in boot.config
    "memorysetup-bucket-allocator-granularity=16"
    "memorysetup-bucket-allocator-bucket-count=8"
    "memorysetup-bucket-allocator-block-size=4194304"
    "memorysetup-bucket-allocator-block-count=1"
    "memorysetup-main-allocator-block-size=16777216"
    "memorysetup-thread-allocator-block-size=16777216"
false was not assigned a value
ERROR: Shader UI/Default shader is not supported on this GPU (none of subshaders/fallbacks are suitable)


```

**Comment by kkoshelev** [comment]
@gOOvER perhaps, you should spend more time analyzing the problem. I bet, you didn’t do clean install on Ubuntu 22. As I mentioned above, your changes to bootstrap URL creates a bug.

I see the same problem with different servers, like Valheim, due to the same URL.

Unless you want your egg only work on your setup, I suggest figuring out the issue.

**Comment by gOOvER** [comment]
> @gOOvER maybe you should spend more time analyzing the problem. I bet, you didn’t do clean install on Ubuntu 22. As I mentioned above, your changes to bootstrap URL creates a bug.
> 
> I see the same problem with different servers, like Valheim, due to the same URL.
> 
> My friend and I spent about 3 hours, figuring out why steamcmd suddenly started crashing on my server for multiple eggs. This url loads outdated binary which tries to bootstrap Steam from url which is no longer available. So I’m fairly certain that isn’t isolated to my server only

What Bootstrap URL? No URL was changed.

The Egg works and only for you Not; then think about, what wrong with your Server.

**Comment by gOOvER** [comment]
@kkoshelev  because you deleted:

`Yes, personal adult entertainment collection is based on unraid. However Pterodactyl is installed on a separate Ubuntu 22.04.1 LTS VM under ESXi. Thanks for asking`

Unraid and esxi are NOT SUPPORTED 
by pterodactyl

No need to disuss Here longer with you

**Comment by Butain** [comment]
This egg is working fine for me in Ubuntu 18.04.5 LTS, Thanks gOovER!

**Comment by kkoshelev** [comment]
> Unraid and esxi are NOT SUPPORTED by pterodactyl

I'm surprised you even know what ESXi is, lol.

Anyhow, I think I found the root cause of the issue. It was the wrong default "domain" setting on my router.

But since you will likely say that pterodactyl doesn't support "default domain" either, I'll include the culprit here for the future generations.

Bootstrap URL
`https://steamcdn-a.akamaihd.net/client/installer/steamcmd_linux.tar.gz` (1) vs 
`http://media.steampowered.com/installer/steamcmd_linux.tar.gz` (2)

(1) results in crash of steamcmd while bootstrapping while (2) doesn't.
Bug #1 -  steamcmd shouldn't crash

During bootstrap, steamcmd binary downloads latest version of steamcmd from `client-download.steampowered.com`.

`client-download.steampowered.com` points to `update.client.steamcontent.akadns.net`
`update.client.steamcontent.akadns.net` is unresolvable
```
$ dig client-download.steampowered.com
;; ANSWER SECTION:
client-download.steampowered.com. 134 IN CNAME	update.client.steamcontent.akadns.net.

$ dig update.client.steamcontent.akadns.net
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1
```

Bug #2 - steamcmd shouldn't request update from invalid domain

On my Ubuntu box, DNS resolver resolves any publicly unresolved domains to WAN IP of the router due to a combination of "default domain" router setting and wildcard DNS setup for that domain. WAN IP serves HTTP page which makes steamcmd crash.

Bug #3 - publicly unresolved domains should not resolve to WAN IP

Fix: Go to `System: Settings: General` in the router and set `Domain` setting to `home.apra`.

**Comment by gOOvER** [comment]
> > Unraid and esxi are NOT SUPPORTED by pterodactyl
> 
> I'm surprised you even know what ESXi is, lol.
> 
> Anyhow, I think I found the root cause of the issue. It was the wrong default "domain" setting on my router.
> 
> But since you will likely say that pterodactyl doesn't support "default domain" either, I'll include the culprit here for the future generations.
> 
> Bootstrap URL `https://steamcdn-a.akamaihd.net/client/installer/steamcmd_linux.tar.gz` (1) vs `http://media.steampowered.com/installer/steamcmd_linux.tar.gz` (2)
> 
> (1) results in crash of steamcmd while bootstrapping while (2) doesn't. Bug #1 - steamcmd shouldn't crash
> 
> During bootstrap, steamcmd binary downloads latest version of steamcmd from `client-download.steampowered.com`.
> 
> `client-download.steampowered.com` points to `update.client.steamcontent.akadns.net` `update.client.steamcontent.akadns.net` is unresolvable
> 
> ```
> $ dig client-download.steampowered.com
> ;; ANSWER SECTION:
> client-download.steampowered.com. 134 IN CNAME	update.client.steamcontent.akadns.net.
> 
> $ dig update.client.steamcontent.akadns.net
> ;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1
> ```
> 
> Bug #2 - steamcmd shouldn't request update from invalid domain
> 
> On my Ubuntu box, DNS resolver resolves any publicly unresolved domains to WAN IP of the router due to a combination of "default domain" router setting and wildcard DNS setup for that domain. WAN IP serves HTTP page which makes steamcmd crash.
> 
> Bug #3 - publicly unresolved domains should not resolve to WAN IP
> 
> Fix: Go to `System: Settings: General` in the router and set `Domain` setting to `home.apra`.

Nothing of them you posted above is egg realated or Steam related.

Egg works fine:

```
container@pterodactyl~ Starting installation process, this could take a few minutes...
steam user is not set.
Using anonymous user.
steamcmd.sh
linux32/steamcmd
linux32/steamerrorreporter
linux32/libstdc++.so.6
linux32/crashhandler.so
Redirecting stderr to '/mnt/server/Steam/logs/stderr.txt'
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[  0%] Checking for available update...
[----] Downloading update (0 of 57385 KB)...
[  0%] Downloading update (0 of 57385 KB)...
[  0%] Downloading update (0 of 57385 KB)...
[  0%] Downloading update (3084 of 57385 KB)...
[  5%] Downloading update (6943 of 57385 KB)...
[ 12%] Downloading update (9405 of 57385 KB)...
[ 16%] Downloading update (11858 of 57385 KB)...
[ 20%] Downloading update (14572 of 57385 KB)...
[ 25%] Downloading update (17042 of 57385 KB)...
[ 29%] Downloading update (19499 of 57385 KB)...
[ 33%] Downloading update (22199 of 57385 KB)...
[ 38%] Downloading update (24650 of 57385 KB)...
[ 42%] Downloading update (27107 of 57385 KB)...
[ 47%] Downloading update (29803 of 57385 KB)...
[ 51%] Downloading update (32249 of 57385 KB)...
[ 56%] Downloading update (34710 of 57385 KB)...
[ 60%] Downloading update (37421 of 57385 KB)...
[ 65%] Downloading update (39863 of 57385 KB)...
[ 69%] Downloading update (41227 of 57385 KB)...
[ 71%] Downloading update (42476 of 57385 KB)...
[ 74%] Downloading update (43832 of 57385 KB)...
[ 76%] Downloading update (45057 of 57385 KB)...
[ 78%] Downloading update (46285 of 57385 KB)...
[ 80%] Downloading update (47637 of 57385 KB)...
[ 83%] Downloading update (48858 of 57385 KB)...
[ 85%] Downloading update (50084 of 57385 KB)...
[ 87%] Downloading update (51434 of 57385 KB)...
[ 89%] Downloading update (52663 of 57385 KB)...
[ 91%] Downloading update (53894 of 57385 KB)...
[ 93%] Downloading update (55234 of 57385 KB)...
[ 96%] Downloading update (56461 of 57385 KB)...
[ 98%] Downloading update (57385 of 57385 KB)...
[100%] Download Complete.
[----] Applying update...
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Cleaning up...
[----] Update complete, launching...
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
Redirecting stderr to '/mnt/server/Steam/logs/stderr.txt'
/tmp/dumps insufficient permissions - delete and recreate
[  0%] Checking for available updates...
[----] Verifying installation...
[  0%] Downloading update...
[  0%] Checking for available updates...
[----] Download complete.
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Cleaning up...
[----] Update complete, launching Steamcmd...
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
Redirecting stderr to '/mnt/server/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation - version 1669935972
-- type 'quit' to exit --
Loading Steam API...OK
Connecting anonymously to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK
 Update state (0x3) reconfiguring, progress: 0.00 (0 / 0)
 Update state (0x3) reconfiguring, progress: 0.00 (0 / 0)
 Update state (0x61) downloading, progress: 4.96 (133376576 / 2686736166)
 Update state (0x61) downloading, progress: 23.02 (618578364 / 2686736166)
 Update state (0x61) downloading, progress: 31.52 (846826110 / 2686736166)
 Update state (0x61) downloading, progress: 42.96 (1154172052 / 2686736166)
 Update state (0x61) downloading, progress: 51.65 (1387661771 / 2686736166)
 Update state (0x61) downloading, progress: 59.26 (1592103695 / 2686736166)
 Update state (0x61) downloading, progress: 68.40 (1837676088 / 2686736166)
 Update state (0x61) downloading, progress: 72.78 (1955359121 / 2686736166)
 Update state (0x61) downloading, progress: 72.78 (1955359121 / 2686736166)
 Update state (0x61) downloading, progress: 78.41 (2106751462 / 2686736166)
 Update state (0x61) downloading, progress: 87.21 (2343047988 / 2686736166)
 Update state (0x61) downloading, progress: 96.19 (2584287882 / 2686736166)
 Update state (0x61) downloading, progress: 96.19 (2584287882 / 2686736166)
 Update state (0x61) downloading, progress: 100.00 (2686736166 / 2686736166)
 Update state (0x61) downloading, progress: 100.00 (2686736166 / 2686736166)
 Update state (0x61) downloading, progress: 100.00 (2686736166 / 2686736166)
 Update state (0x81) verifying update, progress: 3.21 (86214942 / 2686736166)
 Update state (0x1) running, progress: 0.00 (0 / 0)
Success! App '600760' fully installed.
'linux32/steamclient.so' -> '../.steam/sdk32/steamclient.so'
'linux64/steamclient.so' -> '../.steam/sdk64/steamclient.so'
-----------------------------------------
Installation completed...
-----------------------------------------
[Pterodactyl Daemon]: Checking server disk space usage, this could take a few seconds...
[Pterodactyl Daemon]: Updating process configuration files...
[Pterodactyl Daemon]: Ensuring file permissions are set correctly,


```

**Comment by X-Coder** [comment]
Thank you for your updates, this patch worked fine for me a few weeks ago.

Sadly currently is crashing on start on both of my servers. I found multiple users on official discord with the same issue.
Can you please verify this egg is still working for you?

It seems, since stationeers did a new game release, ~two weeks ago, this egg is broken because of a missing dependency, maybe it requires a newer version of glibc. 

From what I have read on the official stationeers discord, the new game release is now based on a newer glibc release because it is based now on ubuntu 22. With Debian 11 it does not work any more.
[The user mike842 confirmed, dedicated server works in debian testing bookworm.](https://discord.com/channels/276525882049429515/392080751648178188/1055378559851569212)

This is the error message, when the server crashes at start:
`DllNotFoundException: RocketNet assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) RocketNet.RocketNetNative.abi_GetInstance()
  at RocketNet.RocketNetInstance..ctor () [0x00006] in <48b2691811284ed4b05f438d7468f1d9>:0 
  at Assets.Scripts.Networking.NetworkManager..cctor () [0x0004e] in <c4203c98d3da42cd97754c092c6c3f1d>:0 
Rethrow as TypeInitializationException: The type initializer for 'Assets.Scripts.Networking.NetworkManager' threw an exception.
  at NetworkBase..cctor () [0x00000] in <c4203c98d3da42cd97754c092c6c3f1d>:0 
Rethrow as TypeInitializationException: The type initializer for 'Assets.Scripts.NetworkServer' threw an exception.
  at Assets.Scripts.GameManager.Awake () [0x0003c] in <c4203c98d3da42cd97754c092c6c3f1d>:0 
UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception, Object)
Assets.Scripts.GameManager:Awake()`

**Pull request merged**